packages/core/src/browser/dialogs.ts

```diff
@@ -194,31 +194,45 @@ export abstract class AbstractDialog<T> extends BaseWidget {
         this.activeElement = undefined;
         super.close();
     }
-
     protected onUpdateRequest(msg: Message): void {
         super.onUpdateRequest(msg);
         this.validate();
     }
 
-    protected validate(): void {
+    protected validateCancellationSource = new CancellationTokenSource();
+    protected async validate(): Promise<void> {
         if (!this.resolve) {
             return;
         }
+        this.validateCancellationSource.cancel();
+        this.validateCancellationSource = new CancellationTokenSource();
+        const token = this.validateCancellationSource.token;
         const value = this.value;
-        const error = this.isValid(value, 'preview');
+        const error = await this.isValid(value, 'preview');
+        if (token.isCancellationRequested) {
+            return;
+        }
         this.setErrorMessage(error);
     }
 
-    protected accept(): void {
-        if (this.resolve) {
-            const value = this.value;
-            const error = this.isValid(value, 'open');
-            if (!DialogError.getResult(error)) {
-                this.setErrorMessage(error);
-            } else {
-                this.resolve(value);
-                Widget.detach(this);
-            }
+    protected acceptCancellationSource = new CancellationTokenSource();
+    protected async accept(): Promise<void> {
+        if (!this.resolve) {
+            return;
+        }
+        this.validateCancellationSource.cancel();
```

Comment:
`acceptCancellationSource` in this method

Reply:
ah, a bad copy and paste 🤣